togetherly/lib/views/widgets/new_chore_dialog.dart

```diff
@@ -0,0 +1,400 @@
+        title: _title,
+        description: _description,
+        points: _points,
+        dueDate: _dueDate,
+        isShared: _isShared,
+        assignedChildId: 1,
+      );
+      setState(() => _loading = true);
+      provider.addChore(newChore);
+      setState(() => _loading = false);
+      Navigator.of(context).pop();
+    }
+  }
+
+  String? validatePoints(String value) {
+    if (value.isEmpty) {
+      setState(() => _points = 0);
+    } else {
+      final numericRegex = RegExp(r'^[0-9]+$');
+      if (numericRegex.hasMatch(value)) {
+        setState(() => _points = int.tryParse(value) ?? 0);
+      } else {
+        return 'Value can only contain numeric characters';
+      }
+    }
+    return null;
+  }
+
+  String prettyDate(String date) {
+    List<String> dateTimeArr = date.split(' ');
+    List<String> dateArr = dateTimeArr.elementAt(0).split('-');
+    return '${dateArr.elementAt(1)}/${dateArr.elementAt(2)}/${dateArr.elementAt(0)}';
+  }
+
+  Widget buildForm(BuildContext context) {
+    return Form(
+      key: _formKey,
+      child: Column(
+        mainAxisSize: MainAxisSize.min,
+        children: <Widget>[
+          TextFormField(
+            controller: _titleController,
+            decoration: const InputDecoration(
+              labelText: 'Chore title',
+            ),
+            onChanged: (value) => setState(() => _title = value),
+            validator: (value) {
+              if (value == null || value.isEmpty) {
+                return 'Please enter a title';
+              }
+              return null;
+            },
+          ),
+          TextFormField(
+            controller: _descriptionController,
+            decoration: const InputDecoration(
+              labelText: 'Chore details',
+            ),
+            onChanged: (value) => setState(() => _description = value),
+          ),
+          const SizedBox(
+            height: 20,
+          ),
+          Row(
+            mainAxisSize: MainAxisSize.max,
+            children: [
+              const Text(
+                'Points:',
+                style: AppTextStyles.brandBody,
+              ),
+              const SizedBox(
+                width: 20,
+              ),
+              SizedBox(
+                width: 50,
+                height: 25,
+                child: TextFormField(
+                  style: AppTextStyles.brandBody,
+                  controller: _pointsController,
+                  keyboardType: TextInputType.number,
+                  onChanged: (value) => validatePoints(value),
+                  validator: (value) {
+                    if (int.tryParse(value!) != null) {
+                      if (int.tryParse(value)! > 100) {
+                        return 'Value cannot be larger than 100';
+                      }
+                    } else {
+                      return 'Value can only contain numeric characters';
+                    }
+                    return null;
+                  },
```

Comment:
It seems like there may be a better way to do this kind of validation: https://stackoverflow.com/questions/49577781/how-to-create-number-input-field-in-flutter

Reply:
Thanks for the tip, I added the inputFormatters line which eliminates the need for the outer if statement in the validator.